src/metabase/events/activity_feed.clj

```diff
@@ -47,99 +50,120 @@
-        #(select-keys % [:description :name])
+  [topic event]
+  (let [dashboard             (:object event)
+        create-delete-details
+        #(select-keys (:object %) [:description :name])
 
         add-remove-card-details
-        (fn [{:keys [dashcards] :as obj}]
-          ;; we expect that the object has just a dashboard :id at the top level
-          ;; plus a `:dashcards` attribute which is a vector of the cards added/removed
-          (-> (t2/select-one [Dashboard :description :name], :id (events/object->model-id topic obj))
-              (assoc :dashcards (for [{:keys [id card_id]} dashcards]
-                                  (-> (t2/select-one [Card :name :description], :id card_id)
-                                      (assoc :id id)
-                                      (assoc :card_id card_id))))))]
+        (fn [{:keys [object dashcards] :as _event}]
+          (let [card-ids             (map :card_id dashcards)
+                card-id->dashcard-id (into {} (map (juxt :card_id :id) dashcards))
+                dashboard            (select-keys object [:name :description])
+                ;; TODO: do we still even use this information? if not let just save dashcards as is
+                dashcards            (if (seq card-ids)
+                                       (->> (t2/select [:model/Card :id :name :description] :id [:in card-ids])
+                                            (map #(set/rename-keys % {:id :card_id}))
+                                            (map #(assoc % :id (get card-id->dashcard-id (:card_id %)))))
+                                       [])]
+            (assoc dashboard :dashcards dashcards)))]
+
     (activity/record-activity!
-      :topic      topic
-      :object     object
-      :details-fn (case topic
-                    :event/dashboard-create       create-delete-details
-                    :event/dashboard-delete       create-delete-details
-                    :event/dashboard-add-cards    add-remove-card-details
-                    :event/dashboard-remove-cards add-remove-card-details))))
+     {:topic    topic
+      :model    "dashboard"
+      :model-id (:id dashboard)
+      :user-id  (:actor-id event)
+      :object   dashboard
+      :details  (case topic
+                  :event/dashboard-create       (create-delete-details event)
+                  :event/dashboard-delete       (create-delete-details event)
+                  :event/dashboard-add-cards    (add-remove-card-details event)
+                  :event/dashboard-remove-cards (add-remove-card-details event))})))
 
 (derive ::metric-event ::event)
 (derive :event/metric-create ::metric-event)
 (derive :event/metric-update ::metric-event)
 (derive :event/metric-delete ::metric-event)
 
 (methodical/defmethod events/publish-event! ::metric-event
-  [topic object]
-  (let [details-fn  #(select-keys % [:name :description :revision_message])
-        table-id    (:table_id object)
+  [topic event]
```

Comment:
Again, I think it would be nicer to destructure all the keys of event here

Reply:
https://github.com/metabase/metabase/pull/34947/commits/d542853d655500e2e2288514f83a1d6ec589179f